src/Domain/Creators/ShapeOptions.hpp

```diff
@@ -0,0 +1,122 @@
+  std::array<double, 3> spin{std::numeric_limits<double>::signaling_NaN(),
+                             std::numeric_limits<double>::signaling_NaN(),
+                             std::numeric_limits<double>::signaling_NaN()};
+};
+
+/// Label for shape map options
+struct Spherical {};
+
+/*!
+ * \brief Class to be used as an option for initializing shape map coefficients.
+ *
+ * \tparam IncludeTransitionEndsAtCube This is mainly added for the
+ * `domain::creators::BinaryCompactObject` domain.
+ * \tparam Object Which object that this shape map represents. Use
+ * `domain::ObjectLabel::None` if there is only a single object in your
+ * simulation.
+ */
+template <bool IncludeTransitionEndsAtCube, domain::ObjectLabel Object>
+struct ShapeMapOptions {
+  using type = Options::Auto<ShapeMapOptions, Options::AutoLabel::None>;
+  static std::string name() { return "ShapeMap" + get_output(Object); }
+  static constexpr Options::String help = {
+      "Options for a time-dependent distortion (shape) map about the "
+      "specified object. Specify 'None' to not use this map."};
+
+  struct LMax {
+    using type = size_t;
+    static constexpr Options::String help = {
+        "LMax used for the number of spherical harmonic coefficients of the "
+        "distortion map."};
+  };
+
+  struct InitialValues {
+    using type =
+        Options::Auto<std::variant<KerrSchildFromBoyerLindquist>, Spherical>;
+    static constexpr Options::String help = {
+        "Initial Ylm coefficients for the shape map. Specify 'Spherical' for "
+        "all coefficients to be initialized to zero."};
+  };
+
+  struct SizeInitialValues {
+    using type = Options::Auto<std::array<double, 3>>;
+    static constexpr Options::String help = {
+        "Initial value and two derivatives of the 00 coefficient. Specify "
+        "'Auto' to use the 00 coefficient specified in the 'InitialValues' "
+        "option."};
+  };
+
+  struct TransitionEndsAtCube {
+    using type = bool;
+    static constexpr Options::String help = {
+        "If 'true', the shape map transition function will be 0 at the cubical "
+        "boundary around the object. If 'false' the transition function will "
+        "be 0 at the outer radius of the inner sphere around the object"};
+  };
+
+  using common_options = tmpl::list<LMax, InitialValues, SizeInitialValues>;
+
+  using options =
+      tmpl::conditional_t<IncludeTransitionEndsAtCube,
+                          tmpl::push_back<common_options, TransitionEndsAtCube>,
+                          common_options>;
+
+  size_t l_max{};
+  std::optional<std::variant<KerrSchildFromBoyerLindquist>> initial_values{};
+  std::optional<std::array<double, 3>> initial_size_values{};
+  bool transition_ends_at_cube{false};
+};
+
+template <bool IncludeTransitionEndsAtCube, domain::ObjectLabel Object>
+std::pair<std::array<DataVector, 3>, std::array<DataVector, 4>>
+construct_initial_shape_and_size_funcs(
```

Comment:
(nitpicking) `construct_` is redundant, we don't add prefixes like `compute_` to functions.